### DISCUSSION TITLE
Use Sonoff RF Bridge to emulate a device

### DISCUSSION THREAD

**Discussion by Rasenkantenstein** [arendst/Tasmota#13676] [Q&A] [answered]
Hi,

I have a Sonoff RF Bridge flashed with Tasmota 10.0. I use this device to control some RF shutters in my house. I used the rf177 command to sniff all the signals from the devices remote control in order to send rfraw commands through Home Assistant. I have 8 shutters and all of them work. But the ninth is a pain. I don't get the correct result to close the shutters. I tried a 100 times...

Now I got the idea to just emulate any RF remote control with 3 buttons. The (really bad) [documentation](https://wts-schaltgeraete.de/wp-content/uploads/Bedienungsanleitungen/Produktgruppe-02%20Funktechnik%20Serie%20DMF/Funk-Wandsender%201-Kanal%20und%202-Kanal%20DMF-WS1%20+%20DMF-WS2.pdf) of the manufacturer presents the necessary steps in order to teach a new sender:

press the close and open buttons simultaneously for at least 2 seconds
press the stop button
press open on the new sender
Would it work to let bridge just be another item? To disguise itself as another generic 3-button remote control with which I can just start the sequence mentioned above?
(Or for that matter, is there a database with the necessary rfraw commands?)

This is the device:

[WTS dmf shutter engine](https://wts-schaltgeraete.de/wp-content/uploads/Bedienungsanleitungen/Produktgruppe-01%20Antriebstechnik%20+%20Zubehör/Rohrmotoren%20Funk%20Serie%20DMF.pdf)
radio frequency 3 button hand receiver

**Comment by barbudor** [comment]
The links you provided all fail. But however if they are in german, I won't be able to understand a single word :)

As far as I know there is no database of RF codes that would allow you to simulate a given RF remote.
You can only learn the codes from listenning to the remote itself

Now if you have learned some codes, it should be possible for you to create new codes that don't interfere with the first and make the 9th controller learn them.

**Reply by Rasenkantenstein** [reply]
Thank you barbudor, unfortunately I don't _really_ understand what's going on with interpreting the codes. The best I could find right now was the wiki on the Portisch wiki... But I am not yet able to understand this. Would you have a book/tutorial/video on hand to point me in the right direction? How can I make new codes?
I am willing to spend several hours to close a shutter with MQTT that I am using only thrice per year...  🤪

Can I make Tasmota send two commands simultaneously or is this also just a single command triggered by pushing to buttons at the same time?

**Comment by Jason2866** [comment, marked as answer]
I have read the manual. It is probably a rolling code which is teached between the devices.
The Sonoff RF MCU is to limited in resources to support rolling codes. Even it would work the used protocol and the biggest challenge the key needs to be known.
Short answer: Not possible
The only way is to use the included rf remotes and make this devices smart -> Controll the buttons
